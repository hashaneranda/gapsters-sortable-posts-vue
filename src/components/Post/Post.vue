<template>
  <div
    class="flex flex-row items-center max-w-md p-6 my-3 transition duration-500 ease-in-out transform bg-white rounded-lg shadow-xl hover:scale-110"
  >
    <h1
      class="flex-grow text-xl leading-tight text-gray-900"
      data-testid="title"
    >
      Post {{ id }}
    </h1>
    <div class="flex-grow-0">
      <img
        v-if="!isStart"
        :src="downArrow"
        alt="arrow"
        class="w-4 mb-5
      transform rotate-180 action__icon"
        data-testid="button-up"
        @click="() => buttonUp(id)"
      />
      <img
        v-if="!isEnd"
        :src="downArrow"
        alt="arrow"
        class="w-4 action__icon"
        data-testid="button-down"
        @click="() => buttonDown(id)"
      />
    </div>
  </div>
</template>

<script>
// import downArrow from "../../assets/images/downArrow.svg";

export default {
  name: "Post",
  props: {
    id: Number,
    isStart: Boolean,
    isEnd: Boolean,
    buttonUp: Function,
    buttonDown: Function,
  },
  computed: {
    downArrow() {
      return require("../../assets/images/downArrow.svg");
    },
  },
};
</script>

<style>
.action__icon:hover {
  filter: brightness(0%);
  cursor: pointer;
}
</style>
