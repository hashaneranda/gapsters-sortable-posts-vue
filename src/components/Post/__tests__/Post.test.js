import { render, cleanup, fireEvent } from "@testing-library/vue";
import { mount } from "@vue/test-utils";

import Post from "../Post";

afterEach(cleanup);

describe("Post Componenet", () => {
  it("should take a snapshot", () => {
    const wrapper = mount(Post);
    expect(wrapper.element).toMatchSnapshot();
  });

  it("title should equal to empty", () => {
    const { getByTestId } = render(Post);
    expect(getByTestId("title")).toHaveTextContent("Post");
  });

  it("should have two buttons", () => {
    const { getByTestId } = render(Post);
    expect(getByTestId("button-up")).toBeInTheDocument();
    expect(getByTestId("button-down")).toBeInTheDocument();
  });

  it("renders componenet corretly without passing argumenets", () => {
    const { getByTestId } = render(Post);
    expect(getByTestId("button-up")).toBeInTheDocument();
    expect(getByTestId("button-down")).toBeInTheDocument();
    expect(getByTestId("title")).toHaveTextContent("Post");
  });

  it("renders componenet corretly with id argumenet", () => {
    const { getByTestId } = render(Post, {
      props: {
        id: 2,
      },
    });
    expect(getByTestId("button-up")).toBeInTheDocument();
    expect(getByTestId("button-down")).toBeInTheDocument();
    expect(getByTestId("title")).toHaveTextContent("Post 2");
  });

  it("renders componenet corretly with id argumenet, buttonUp argument, buttonDown argument", () => {
    const testFunction = () => {
      testVal = true;
    };

    const { getByTestId } = render(Post, {
      props: {
        id: 3,
        buttonUp: testFunction,
        buttonDown: testFunction,
      },
    });

    // const { getByTestId } = render(
    //   <Post id="3" buttonUp={testFunction} buttonDown={testFunction} />
    // );
    expect(getByTestId("button-up")).toBeInTheDocument();
    expect(getByTestId("button-down")).toBeInTheDocument();
    expect(getByTestId("title")).toHaveTextContent("Post 3");
  });

  it("Both up button and down button function calls working on button click", () => {
    let testValUp = false;
    let testValDown = false;

    const testFunctionUp = () => {
      testValUp = true;
    };

    const testFunctionDown = () => {
      testValDown = true;
    };

    const { getByTestId } = render(Post, {
      props: {
        id: 3,
        buttonUp: testFunctionUp,
        buttonDown: testFunctionDown,
      },
    });

    fireEvent.click(getByTestId("button-up"));
    fireEvent.click(getByTestId("button-down"));

    expect(getByTestId("button-up")).toBeInTheDocument();
    expect(getByTestId("button-down")).toBeInTheDocument();
    expect(getByTestId("title")).toHaveTextContent("Post 3");

    expect(testValUp).toEqual(true);
    expect(testValDown).toEqual(true);
  });
});
