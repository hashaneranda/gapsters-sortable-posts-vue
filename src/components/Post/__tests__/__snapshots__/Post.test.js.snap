// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post Componenet should take a snapshot 1`] = `
<div
  class="flex flex-row items-center max-w-md p-6 my-3 transition duration-500 ease-in-out transform bg-white rounded-lg shadow-xl hover:scale-110"
>
  <h1
    class="flex-grow text-xl leading-tight text-gray-900"
    data-testid="title"
  >
    
    Post 
  
  </h1>
   
  <div
    class="flex-grow-0"
  >
    <img
      alt="arrow"
      class="w-4 mb-5
    transform rotate-180 action__icon"
      data-testid="button-up"
      src=""
    />
     
    <img
      alt="arrow"
      class="w-4 action__icon"
      data-testid="button-down"
      src=""
    />
  </div>
</div>
`;
